### PR TITLE
DYN-6036: Set dropdown IsModified outside WPF

### PR DESCRIPTION
**Pull request by twastvedt**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

[DYN-6036](https://jira.autodesk.com/browse/DYN-6036)

Moves dropdowns' OnNodeModified call outside WPF so that it is also run in ui-less mode.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

Fix a bug where dropdown nodes were not marked as modified when the node's value was changed programatically while running in ui-less mode.

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

**Comment by QilongTang** [comment]
Looks safe but just curious have you tested in Revit that it does not affect Dropdown nodes?

**Comment by twastvedt** [comment]
@QilongTang Just did, Revit dropdowns look to be working as usual.

File: test/DynamoCoreTests/Nodes/CustomDropdownTests.cs [hunk 1]
```diff
@@ -40,12 +38,22 @@ public void OpenJsonDYNWithCorrectMenuItems()
...
+        }
 
-            Graph.Nodes.NodeModel node = CurrentDynamoModel.CurrentWorkspace.Nodes.FirstOrDefault();
+        [Test]
+        public void OpenJsonDYNWithCorrectMenuItems()
+        {
+            LoadSamplesPackage();
+            RunDropdownGraph();
+
+            NodeModel node = CurrentDynamoModel.CurrentWorkspace.Nodes.FirstOrDefault();
```
Comment (mjkkirschner) [inline test/DynamoCoreTests/Nodes/CustomDropdownTests.cs:56]: Is there only one node in this graph?

**Comment by mjkkirschner** [comment]
I am curious did you just run into this or is it a longstanding bug?

**Comment by twastvedt** [comment]
> I am curious did you just run into this or is it a longstanding known issue?

Just ran into it, though I think it's been around, undiscovered. When Player runs graphs it always forces Dynamo to reload the graph - a design decision. In testing the new player we forgot about that detail and were running without reloading, seeing dropdowns not rerunning their AST when the value was changed.

File: test/DynamoCoreTests/Nodes/CustomDropdownTests.cs (hunk 1 above)
Comment (twastvedt) [reply test/DynamoCoreTests/Nodes/CustomDropdownTests.cs:56]: Two nodes. The first is the dropdown, the second is a watch node.

**Review by QilongTang — APPROVED** [review]

**Review by mjkkirschner — APPROVED** [review]

**Pull request merged**